chapters/intro-computer-architecture/binary-hex/reading/README.md

```diff
@@ -0,0 +1,142 @@
+#### Logical Shifts
+
+Logical shifts left/right involve moving each bit by one position.
+Since the result must be on the same number of bits as the initial value, the first bit is lost, and the empty space is filled with a 0 bit.
+
+![Left Logical Shift](../media//left-logical-shift.svg)
+
+![Right Logical Shift](../media//right-logical-shift.svg)
+
+For explanations related to bitwise operations in C, refer to the guide at [Bitwise Operators in C](https://www.geeksforgeeks.org/bitwise-operators-in-c-cpp/).
+
+## Hexadecimal System
+
+In the **hexadecimal system** (base 16), values are represented as a string of characters from '0' to '9' or 'a' to 'f'.
+A byte consists of two such characters, so each character corresponds to a group of 4 bits (a nibble).
+
+### Conversion from Decimal to Binary/Hexadecimal
+
+- Divide the number successively by the base number (2 or 16) and keep the remainders.
+- When the quotient of the division becomes 0, write down the remainders in reverse order.
+- In the case of base 16, when the remainder is greater than 9, letters a-f are used (`0xa = 10`, `0xf = 15`).
+
+#### Example: Conversion of the number 0xD9B1 to decimal
+
+$$ \texttt{0xD9B1} = 1 \cdot 16 ^ 0 + 11 \cdot 16 ^ 1 + 9 \cdot 16 ^ 2 + 13 \cdot 16 ^ 3 = 55729 $$
+
+### Conversion between Binary and Hexadecimal
+
+As mentioned earlier, a digit in a hexadecimal number corresponds to a group of 4 bits (a nibble).
+Therefore, to convert a number from hexadecimal to binary, it's sufficient to transform each digit into the equivalent 4-bit group.
+
+#### Example: Conversion of the number 0xD9B1 to binary
+
+- `0x1 = 0b0001`
+- `0xB = 0b1011`
+- `0x9 = 0b1001`
+- `0xD = 0b1101`
+
+Thus, the resulting binary number is `0b1101100110110001`.
+
+The reverse operation, conversion from binary to hexadecimal, can be done by converting each group of 4 bits into the corresponding digit in hexadecimal.
+
+### Use of Base 16 Representation
+
+The hexadecimal system is used to represent memory addresses and to visualize data in a more interpretable way than a sequence composed only of 0s and 1s.
+The image below provides an example in this regard:
+
+![Memory Map](../media/hex-view.png)
+
+(Image taken from [Digital Detective](https://www.digital-detective.net/))
+
+## Representation of Data Types
+
+In a computer's memory, a value is stored on a fixed number of bits.
+Depending on the architecture, each processor can access a maximum number of bits in a single operation, which represents the word size.
+
+The sizes of common data types used in C are dependent on both the processor and the platform on which the program was compiled (operating system, compiler).
+The table below presents the sizes of data types on a 32-bit architecture processor, when the program is compiled using gcc under Linux.
+
+On the left side, we have memory addresses where data is located. At address `0x0009FA08`, the first 4 bytes starting from offset `0x02` are `0x01` `0x00`, `0xFF`, `0xFF`.
```

Comment:
Does this reference the image from "Use of Base 16 Represenetation"? If so, I think the image / sub-heading could be moved here, to "Representation of Data Types" or something else. My issue is that it's not clear what this "left side" is referencing.

Reply:
I've added a link to it. It should suffice.